tests/libmonitoring/metric_matcher.go

```diff
@@ -0,0 +1,75 @@
+package libmonitoring
+
+import (
+	"fmt"
+
+	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
+	"github.com/onsi/gomega/format"
+)
+
+const (
+	prometheusMetricNameLabel       = "__name__"
+	prometheusHistogramBucketPrefix = "_bucket"
+)
+
+type MetricMatcher struct {
+	Metric operatormetrics.Metric
+	Labels map[string]string
+}
+
+func (matcher *MetricMatcher) FailureMessage(actual interface{}) (message string) {
+	msg := format.Message(actual, "to contain metric", matcher.Metric.GetOpts().Name)
+
+	if matcher.Labels != nil {
+		msg += fmt.Sprintf(" with labels %v", matcher.Labels)
+	}
+
+	return msg
+}
+
+func (matcher *MetricMatcher) NegatedFailureMessage(actual interface{}) (message string) {
```

Comment:
why we need both functions? FailureMessage and NegatedFailureMessage?
seems like they do the same thing except the msg

Reply:
this is a implementation of a gomega interface, so it's mandatory to have both, to handle 2 scenarios

**FailureMessage** is used when you expect to find something and it is not present, for example, **expected [1,2] to contain 3**

**NegatedFailureMessage** is used when you expect to not find something,  but it is there, for example, **expected [A,B] not to contain A**

Reply:
+1 thanks!